### PR TITLE
chore: enforce file extensions on imports

### PR DESCRIPTION
**Pull request by jackfranklin**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -88,7 +89,8 @@ module.exports = {
 
         "no-restricted-imports": ["error", {
             patterns: ["*Events"],
-        }]
+        }],
+        "import/extensions": ["error", "ignorePackages"]
```
Comment (jackfranklin) [inline .eslintrc.js:93]: we need to figure out how packages work for a browser ESM build - but that's a separate issue to this one. I'll have a PR/issue up describing our potential approach later :)

**Review by paullewis — APPROVED** [review]

File: scripts/ensure-correct-devtools-protocol-package.ts [hunk 2]
```diff
@@ -33,10 +33,10 @@
  * find the one closest to our Chromium revision.
  */
 
-import { PUPPETEER_REVISIONS } from '../src/revisions';
+import { PUPPETEER_REVISIONS } from '../src/revisions.js';
 import { execSync } from 'child_process';
 
-import packageJson from '../package.json';
+import packageJson from '../package.json.js';
```
Comment (TimvdLippe) [inline scripts/ensure-correct-devtools-protocol-package.ts]: I think this needs to remain `package.json`

File: test/jshandle.spec.ts [hunk 3]
```diff
@@ -14,13 +14,13 @@
  * limitations under the License.
  */
 
-import expect from 'expect';
+import expect from 'expect.js';
```
Comment (TimvdLippe) [inline test/jshandle.spec.ts]: This should remain `'expect'` as it uses the Node resolution algorithm.

File: test/launcher.spec.ts [hunk 4]
```diff
@@ -16,18 +16,18 @@
 import fs from 'fs';
 import os from 'os';
 import path from 'path';
-import sinon from 'sinon';
+import sinon from 'sinon.js';
```
Comment (TimvdLippe) [inline test/launcher.spec.ts]: Same here.

**Review by TimvdLippe — CHANGES_REQUESTED** [review]

**Pull request merged**